apps/prairielearn/src/components/QuestionsTable.html.ts

```diff
@@ -0,0 +1,220 @@
+import { html, HtmlSafeString } from '@prairielearn/html';
+import { nodeModulesAssetPath } from '../lib/assets';
+import { EncodedData } from '@prairielearn/browser-utils';
+import { CourseInstance } from '../lib/db-types';
+import { QuestionsTableData } from '../models/questions';
+
+export function QuestionsTable({
+  questions,
+  showAddQuestionButton,
+  current_course_instance,
+  course_instances,
+  urlPrefix,
+  plainUrlPrefix,
+  __csrf_token,
+  errorMessage,
+}: {
+  questions: QuestionsTableData[];
+  showAddQuestionButton: boolean;
+  current_course_instance: CourseInstance;
+  course_instances: CourseInstance[];
+  urlPrefix: string;
+  plainUrlPrefix: string;
+  __csrf_token: string;
+  errorMessage: HtmlSafeString;
+}): HtmlSafeString {
+  const has_legacy_questions = questions.some((row) => row.display_type !== 'v3');
+  const course_instance_ids = (course_instances || []).map((course_instance) => course_instance.id);
+  return html`
+    ${EncodedData(
+      { course_instance_ids, showAddQuestionButton, urlPrefix, plainUrlPrefix },
+      'questions-table-data',
+    )}
+    <style>
+      .sticky-column {
+        position: sticky;
+        left: 0;
+        background: white;
+        background-clip: padding-box;
+        box-shadow: inset -1px 0 #dee2e6;
+      }
+      .table-hover tbody tr:hover td.sticky-column {
+        color: #212529;
+        background-color: #efefef;
+      }
+      .fixed-table-toolbar {
+        padding: 0 1em 0 1em;
+      }
+      .fixed-table-toolbar div.pagination,
+      .fixed-table-toolbar div.pagination-detail {
+        margin: 0 1em 0 0 !important;
+      }
+    </style>
+
+    <script src="${nodeModulesAssetPath('bootstrap-table/dist/bootstrap-table.min.js')}"></script>
+    <script src="${nodeModulesAssetPath(
+        'bootstrap-table/dist/extensions/sticky-header/bootstrap-table-sticky-header.min.js',
+      )}"></script>
+    <script src="${nodeModulesAssetPath(
+        'bootstrap-table/dist/extensions/filter-control/bootstrap-table-filter-control.min.js',
+      )}"></script>
+    <link
+      href="${nodeModulesAssetPath('bootstrap-table/dist/bootstrap-table.min.css')}"
+      rel="stylesheet"
+    />
+    <link
+      href="${nodeModulesAssetPath(
+        'bootstrap-table/dist/extensions/sticky-header/bootstrap-table-sticky-header.min.css',
+      )}"
+      rel="stylesheet"
+    />
+    <link
+      href="${nodeModulesAssetPath(
+        'bootstrap-table/dist/extensions/filter-control/bootstrap-table-filter-control.min.css',
+      )}"
+      rel="stylesheet"
+    />
```

Comment:
You should be able to import these into the asset.

Reply:
done for CSS, JS problems discussed in another thread

Reply:
done